File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int n = nums.size();
        int maxsum = 0;
        int ans = INT_MIN;
        for(int i = 0 ; i< n ; i++){
            maxsum += nums[i];
            ans = max(ans,maxsum);
            if(maxsum<0)maxsum = 0;
        }
        return ans;
    }
};

File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        summ = 0
        maxsum = -2**31
        
        if len(nums) == 1:
            return nums[0]
        
        for i in range(len(nums)):
            summ += nums[i]
            maxsum = max(maxsum,summ)
            if(summ < 0):
                summ = 0

        return maxsum 
        